Fix bug with makefile after make test has been run

robber-m opened the pull request:
- grep command was malfunctioning once the file test_ce had been created
- Generalize make test to create a test executable and run it for each
  file named test_<module>.c . For example: test_ce_vim.c would build
  test_ce_vim and link against ce_vim.o
- use the pre-built object file from make when running make test. don't
  rebuild it if you don't need to

bobbymcshane commented on ce.c:
```diff
@@ -432,7 +480,7 @@ bool ce_draw_buffer(const Buffer* buffer, const Point* term_top_left, const Poin
 
-          int64_t min = g_terminal_dimensions->x < line_length ? g_terminal_dimensions->x : line_length;
+          int64_t min = max_width < line_length ? max_width : line_length;
           memset(line_to_print, 0, min + 1);
```
What are you doing with this memset?

robber-m committed:
Fix bug with makefile after make test has been run

- grep command was malfunctioning once the file test_ce had been created
- Generalize make test to create a test executable and run it for each
  file named test_<module>.c . For example: test_ce_vim.c would build
  test_ce_vim and link against ce_vim.o
- use the pre-build object file from make when running make test. don't
  rebuild it if you don't need to

robber-m committed:
Add install target to install ce once it has been build into DESTDIR (overridable variable)

- NOTE: install does not depend on ce because you may have to run it as
  a different user when you go to install it and you may not necessarily
  want to build your executable and object files as that user

justy989 reviewed (approved):
This is moving towards something more complicated than I think I would like
